### DISCUSSION TITLE
connection issues

### DISCUSSION THREAD

**Discussion by d2head** [shupershuff/Diablo2RLoader#37] [General]
I ran it two times without issues and was able to load my accounts and then went to load it again a third time and started having issues with the internet connection. When you go from the D2 screen the connecting to battle net it takes forever, like 3 minutes. Then when you finally get into the game it has you offline and when you click online it says connecting to battle net for like 8 minutes then finally says cannot connect to battle net please make sure you're connected to the internet. Does anyone have any ideas on what is going wrong because i can connect to battle net fine with the diablo battle net launcher.

**Comment by d2head** [comment]
![Screenshot 2024-05-14 182953](https://github.com/shupershuff/Diablo2RLoader/assets/169856412/915f4762-d51b-4131-98ef-44123e0a663f)

**Comment by shupershuff** [comment]
Does the account in question have MFA enabled on it? If so, this doesn't work with parameter based authentication, you would need to either disable MFA or use the Auth Token method instead (outlined in [setup your accounts](https://github.com/shupershuff/Diablo2RLoader?tab=readme-ov-file#3-setup-your-accounts)).

That said, that shouldn't cause the errors shown in the screenshot you've provided. Does this error happen on all of your accounts or just one of them?

**Comment by buggsi** [comment]
Battlenet often wants me to solve a captcha, so the normal user/password authentication fails. However, I set up the token authentication and it works every time.

**Comment by shupershuff** [comment]
That's good. I'm still very curious what caused the error in your screenshot above. I had tried to design the script to output more friendly information for errors I could foresee happening but it seems it could use some improvement. I need to see the first errors to understand what went wrong. I see you've taken a video and screenshot it (I'm guessing because the error was only shown on screen for a short moment), would I be able to get a copy of the video?

Or better yet, in the script if you replace the Clear-Host commands with #Clear-Host, add "Pause" onto line 2568, run this from an already opened PowerShell window, we should be able to see all errors/warnings. You'd need to run the script with your original settings for it to work (ie, temporarily change "Token" back to "Parameter" in the AuthenticationMethod column in accounts.csv)

**Comment by shupershuff** [comment]
Closed discussion. OP's issue is prevented from happening to users in latest version.
